packages/i18n-calypso/src/i18n.js

```diff
@@ -172,16 +168,38 @@ I18N.prototype.emit = function ( ...args ) {
 
 /**
  * Formats numbers using locale settings and/or passed options.
- * @param   {string|number}  number to format (required)
+ * @param   {string | number}  number to format (required)
  * @param   {number | Object}  options  Number of decimal places or options object (optional)
- * @returns {string}         Formatted number as string
+ * @returns {string | number}  Formatted number as string, or original number if formatting fails
  */
 I18N.prototype.numberFormat = function ( number, options = {} ) {
 	const decimals = typeof options === 'number' ? options : options.decimals || 0;
-	const decPoint = options.decPoint || this.state.numberFormatSettings.decimal_point || '.';
-	const thousandsSep = options.thousandsSep || this.state.numberFormatSettings.thousands_sep || ',';
+	const browserSafeLocale = this.getBrowserSafeLocale();
+
+	try {
+		return Intl.NumberFormat( browserSafeLocale, {
+			minimumFractionDigits: decimals, // default is 0
+			maximumFractionDigits: decimals, // default is the greater between minimumFractionDigits and 3
+			// TODO clk numberFormat this may be the only difference, where some cases use 2 (they can just pass the option to Intl.NumberFormat)
```

Comment:
I'm leaving some TODOs with`clk`/initials. I intend to immediately come back to and address in next 1-2 PRs.

Reply:
Something to think about: the `Intl.NumberFormat` is designed to have two steps: 1. create the formatter (once); 2. use it to format numbers (many times).

It's like if creating the formatter was expensive: find the translation resources and initialize everything. And the actual formatting is cheap.

Can we somehow cache the formatter? Create it just once and then in `numberFormat` just call its `.format` method?

Reply:
Thanks. I like the idea, and it's definitely doable. If we keep `numberFormat` going forward, we can explore a caching layer, assuming it brings any real benefits I guess.

It might be the case that we also expose a hook around this.

For now, I think we can proceed with the current form and rework it as we go. WDYT? I intend to have a few more PRs before we make the final call, whether to deprecate or not (and from there do the additional cleanup).

Reply:
Thanks again @jsnajdr. For all your suggestions/guideance. Following up on the above in https://github.com/Automattic/wp-calypso/pull/99110 If my tests are correct, there are indeed gains from a bit of caching.